### ISSUE TITLE
[BUG] `jax.vjp` of a qnode is broken for all but scalar vectors (aka, for the grad case)

### ISSUE THREAD

**Issue by PhilipVinc** [PennyLaneAI/pennylane#3480]
This is a test program I'm trying to fix.

```python
from jax.config import config
config.update("jax_enable_x64", True)

import jax
import jax.numpy as jnp
import numpy as np
import pennylane as qml

phys_qubits = 2
n_configs = 5
np.random.seed(1)
pars_q      = np.random.rand(n_configs,2)
pars_q1      = np.random.rand(2, n_configs)

dev = qml.device("default.qubit", wires=tuple(range(phys_qubits)), shots=None)

def minimal_circ(params):
    @qml.qnode(dev, interface="jax-jit",diff_method="parameter-shift", cache=None)
    def _measure_operator():
        qml.RY(params[...,0],wires=0)
        qml.RY(params[...,1],wires=1)

        op = qml.Hamiltonian([1.0],[qml.PauliZ(0) @ qml.PauliZ(1)])
        return qml.expval(op)
    res = _measure_operator()
    return res

res, vjpfun = jax.vjp(minimal_circ, pars_q)
vjpfun(jnp.ones_like(res))

```

**Comment by CatalinaAlbornoz** [comment]
Hi @PhilipVinc, thank you for pointing this out. We will look into this. How urgently would you need a fix?

**Comment by PhilipVinc** [comment]
I dont _need_ a fix, as today I managed to identify the cause and fix it (I urgently need this so I descended into the depths of Pennylane...)

The issue is quite deep, however. In short (I'm on my phone) it seems to me that the custom_vjp rules in the jax interface are designed to work with a `v` (the vector in vector-jacobian product) that is a scalar, but they don't consider the case where v is a vector.

Fixing this, interestingly, also fixes `jax.jacrev` and `jax.jacobian` and almost all the rest of the jax interface!

I can share a branch that (brutally) addresses the issue but I won't have time to clean it up and make it PR worthy in the next few weeks.

**Comment by CatalinaAlbornoz** [comment]
Wow, thank you @PhilipVinc ! If you could share that branch it could help us look into the issue too.

**Comment by PhilipVinc** [comment]
You can have a look at this commit on the old interface https://github.com/PennyLaneAI/pennylane/compare/master...PhilipVinc:pennylane:pv/vjp2

**Comment by CatalinaAlbornoz** [comment]
Thank you!

**Comment by rmoyard** [comment]
Hi @PhilipVinc! Thanks for the report, I am working on fixing this bug for the new interface (activate it qml.enable_return()). It will produce correct results for broadcasting with one measurement and custom gradients. We plan on making broadcasting with multiple measurements work next year cc @dwierichs .

**Comment by albi3ro** [comment]
This is now fixed as long as you use `qml.transforms.broadcast_expand` to get parameter broadcasting working with the `"parameter-shift"`.  So I'm going to mark this as done.
